homeassistant/components/smartthings/number.py

```diff
@@ -217,3 +249,58 @@ async def async_set_native_value(self, value: float) -> None:
             Command.SET_COOLING_SETPOINT,
             int(value),
         )
+
+
+class SmartThingsFsvSettings(SmartThingsEntity, NumberEntity):
+    """Define a SmartThings FSV setting number."""
+
+    editable = True
+
+    def __init__(
+        self, client: SmartThings, device: FullDevice, component: str, fsv_setting: dict
+    ) -> None:
+        """Initialize the instance."""
+        super().__init__(
+            client,
+            device,
+            {Capability.SAMSUNG_CE_EHS_FSV_SETTINGS},
+            component=component,
+        )
+        self._fsv_id = fsv_setting[FsvSettingProperty.ID]
+        self._attr_unique_id = f"{device.device.device_id}_{component}_{Capability.SAMSUNG_CE_EHS_FSV_SETTINGS}_{Attribute.FSV_SETTINGS}_{self._fsv_id}"
+
+        if fsv_setting[FsvSettingProperty.TYPE] == FsvSettingType.TEMPERATURE:
+            self._attr_native_unit_of_measurement = UNIT_MAP[
+                fsv_setting[FsvSettingProperty.TEMPERATURE_UNIT]
+            ]
+            self._attr_device_class = NumberDeviceClass.TEMPERATURE
+
+        if self._fsv_id in FSV_LABEL_MAP:
+            self._attr_translation_key = f"fsv_setting_{self._fsv_id}"
+        else:
+            self._attr_translation_key = "fsv_setting"
+            self._attr_translation_placeholders = {"fsv_id": str(self._fsv_id)}
+        self._attr_native_min_value = fsv_setting[FsvSettingProperty.MIN_VALUE]
+        self._attr_native_max_value = fsv_setting[FsvSettingProperty.MAX_VALUE]
+        self._attr_step = fsv_setting[FsvSettingProperty.RESOLUTION]
+        self._attr_entity_category = EntityCategory.CONFIG
```

Comment:
Okay so if you look at other platforms like sensor, I think this list of FSV settings would work well in a list of entity descriptions. So that we have a mapping of FSV number -> entity description.